ReactAndroid/src/main/java/com/facebook/react/bridge/BaseJavaModule.java

```diff
@@ -49,17 +54,35 @@
  */
 public abstract class BaseJavaModule implements NativeModule {
   // taken from Libraries/Utilities/MessageQueue.js
-  static final public String METHOD_TYPE_ASYNC = "async";
-  static final public String METHOD_TYPE_PROMISE= "promise";
+  private static final String METHOD_TYPE_ASYNC = "async";
+  private static final String METHOD_TYPE_PROMISE= "promise";
   static final public String METHOD_TYPE_SYNC = "sync";
+  private final List<? extends ArgumentExtractor.Factory> mArgumentExtractorFactories;
+
+  public BaseJavaModule() {
+    this(Collections.<ArgumentExtractor.Factory>emptyList());
+  }
 
-  private static abstract class ArgumentExtractor<T> {
+  public BaseJavaModule(List<? extends ArgumentExtractor.Factory> argumentExtractorFactories) {
```

Comment:
I think I'd prefer having this be provided with from a method instead? 
Like: `List<ArgumentExtractor> getCustomExtractors()`

Also define an extra class so we can tell the type from the extractor, something like:
```
abstract class CustomArgumentExtractor<T> extends ArgumentExtractor {
  Class<T> getSupportedType();
}
```